neighsyncd/neighsync.cpp

```diff
@@ -73,9 +73,6 @@ void NeighSync::onMsg(int nlmsg_type, struct nl_object *obj)
     key+= ":";
 
     nl_addr2str(rtnl_neigh_get_dst(neigh), ipStr, MAX_ADDR_SIZE);
-    /* Ignore IPv6 link-local addresses as neighbors */
```

Comment:
This should be done based on ipv6_link_local_mode

Reply:
@prsunny To make this based on ipv6_link_local_mode, I have to add code to listening to the CONFIG_DB Interface Table ("INTERFACE:Ethernet11") notifications in neighsync code to get the ipv6_link_local_mode right ?

Reply:
Since link-local addresses can be used even when global ipv6 address is configured, we should learn the link-local neighbors in both scenarios.

Reply:
Does this mean, there would be lots of IPv6 linklocal updates to orchagent? I thought, this would be done based on global config.

Reply:
IPv6 link local neighbor updates will come to orchagent only for the interfaces where ipv6 address is configured on it. I mean we have to allow IPv6 LL neighbor when ipv6 address is configured on an interface.

Reply:
I see lots of LL updates to swss record. Can you please fix the [arp_update script](https://github.com/Azure/sonic-buildimage/blob/master/files/scripts/arp_update#L57) to cover link-local?

Reply:
@AkhileshSamineni , tested this change in a device and observed that its now creating LL neighbor and nexthop:

```
Jul 30 19:28:29.107913 NOTICE swss#orchagent: :- addNeighbor: Created neighbor ip fe80::9a03:9bff:fe03:2216, 98:03:9b:03:22:16 on Vlan1000
Jul 30 19:28:29.109199 NOTICE swss#orchagent: :- addNextHop: Created next hop fe80::9a03:9bff:fe03:2216 on Vlan1000
```

```
127.0.0.1:6379[1]> hgetall "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP:oid:0x400000000074b"
1) "SAI_NEXT_HOP_ATTR_TYPE"
2) "SAI_NEXT_HOP_TYPE_IP"
3) "SAI_NEXT_HOP_ATTR_IP"
4) "fe80::9a03:9bff:fe03:2216"
5) "SAI_NEXT_HOP_ATTR_ROUTER_INTERFACE_ID"
6) "oid:0x6000000000662"

1) "ASIC_STATE:SAI_OBJECT_TYPE_NEIGHBOR_ENTRY:{\"ip\":\"fe80::9a03:9bff:fe03:2216\",\"rif\":\"oid:0x6000000000662\",\"switch_id\":\"oid:0x21000000000000\"}"
```

Why is both Neighbor and Nexthop entries created? As per our understanding, LL entries would only be Nexthop in H/W.  @venkatmahalingam cc

Reply:
The entry creation is part of arp_update script ??

Reply:
No, arp_update script is only for future refresh. This is in the current image if dont ignore netlink msg in neighsyncd and neighorch creates the two entries. You should be observing the same when you've a linklocal neighbor right?

Reply:
Yes, Observed the linklocal Neighbor and Nexthop entries 
```
root@sonic:/home/admin# sonic-db-cli ASIC_DB keys \*NEIGH\*
ASIC_STATE:SAI_OBJECT_TYPE_NEIGHBOR_ENTRY:{"ip":"fe80::82a2:35ff:fe32:50be","rif":"oid:0x6000000000b09","switch_id":"oid:0x21000000000000"}
root@sonic:/home/admin# 
root@sonic:/home/admin# 
root@sonic:/home/admin# sonic-db-cli ASIC_DB hgetall "ASIC_STATE:SAI_OBJECT_TYPE_NEXT_HOP:oid:0x4000000000b0a"
SAI_NEXT_HOP_ATTR_IP
fe80::82a2:35ff:fe32:50be
SAI_NEXT_HOP_ATTR_TYPE
SAI_NEXT_HOP_TYPE_IP
SAI_NEXT_HOP_ATTR_ROUTER_INTERFACE_ID
oid:0x6000000000b09
root@sonic:/home/admin#
```

Reply:
Added changes to skip the ipv6 link-local netlink msg in neighsyncd when link-local-mode is disabled.